app/src/main/java/com/faskn/app/weatherapp/di/NetworkModule.kt

```diff
@@ -7,75 +7,69 @@ import com.faskn.app.weatherapp.core.Constants
 import com.faskn.app.weatherapp.domain.DefaultRequestInterceptor
 import com.faskn.app.weatherapp.domain.WeatherAppAPI
 import com.squareup.moshi.Moshi
-import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import dagger.Module
 import dagger.Provides
-import java.util.concurrent.TimeUnit
-import javax.inject.Named
-import javax.inject.Singleton
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
 import okhttp3.Cache
 import okhttp3.OkHttpClient
 import retrofit2.Retrofit
 import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
 import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Named
+import javax.inject.Singleton
 
+@InstallIn(SingletonComponent::class)
 @Module
-class NetModule {
-    @Singleton
+object NetworkModule {
+
     @Provides
+    @Singleton
     @Named("cached")
-    fun provideOkHttpClient(): OkHttpClient {
-        val cache = Cache(Environment.getDownloadCacheDirectory(), 10 * 1024 * 1024)
-        return OkHttpClient.Builder()
+    fun provideOkHttpClient(): OkHttpClient =
+        OkHttpClient.Builder()
             .addNetworkInterceptor(StethoInterceptor())
             .addInterceptor(DefaultRequestInterceptor())
             .readTimeout(1, TimeUnit.MINUTES)
             .writeTimeout(1, TimeUnit.MINUTES)
-            .cache(cache)
+            .cache(Cache(Environment.getDownloadCacheDirectory(), 10 * 1024 * 1024))
             .build()
-    }
 
-    @Singleton
     @Provides
+    @Singleton
     @Named("non_cached")
-    fun provideNonCachedOkHttpClient(): OkHttpClient {
-        return OkHttpClient.Builder()
+    fun provideNonCachedOkHttpClient(): OkHttpClient =
+        OkHttpClient.Builder()
```

Comment:
Would it make sense if we provide `OkHttpClient.Builder` instance separately but applying `cache` if required otherwise only calling `build` function?

Reply:
I don't know if I understand correctly. We can provide an instance of `OkHttpClient.Builder` and implemented it possible to include `cache` if necessary when building `Retrofit`

9a9a362

Reply:
OK, this looks good if we didn't use `non_cached` `OkHttp` at all, it is safe to remove it IMO.